src/client/src/Client.c

```diff
@@ -862,13 +865,66 @@ STATUS putKinesisVideoFragmentMetadata(STREAM_HANDLE streamHandle, PCHAR name, P
     return retStatus;
 }
 
+/**
+ * Puts an event & associated custom data into the Kinesis Video stream.
+ */
+
+STATUS putKinesisVideoEventMetadata(STREAM_HANDLE streamHandle, UINT32 event, PStreamEventMetadata pStreamEventMetadata)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT8 iter = 0;
+    BOOL releaseClientSemaphore = FALSE, releaseStreamSemaphore = FALSE;
+    PKinesisVideoStream pKinesisVideoStream = FROM_STREAM_HANDLE(streamHandle);
+
+    DLOGS("Put event into an Kinesis Video stream.");
+
+    CHK(pKinesisVideoStream != NULL && pKinesisVideoStream->pKinesisVideoClient != NULL, STATUS_NULL_ARG);
+    if (pStreamEventMetadata != NULL) {
+        CHK(pStreamEventMetadata->numberOfPairs <= MAX_EVENT_CUSTOM_PAIRS, STATUS_INVALID_ARG);
+        for (iter = 0; iter < pStreamEventMetadata->numberOfPairs; iter++) {
+            CHK(pStreamEventMetadata->names[iter] != NULL && pStreamEventMetadata->values[iter] != NULL, STATUS_NULL_ARG);
+        }
+        pStreamEventMetadata->version = STREAM_EVENT_METADATA_CURRENT_VERSION;
+    }
+
+    // check for valid events
+    if (event >= STREAM_EVENT_TYPE_LAST || event == STREAM_EVENT_TYPE_NONE) {
+        CHK(FALSE, STATUS_INVALID_ARG);
+    }

```

Comment:
This check could be before the `if (pStreamEventMetadata != NULL)` to bail out early right after the null checks.

Reply:
It could be, but those lines are also checking the validity of the parameters, so the ordering is pretty insignificant.